proxy_network.c

```diff
@@ -767,6 +783,44 @@ static void _backend_failed(struct mcp_backendconn_s *be) {
     }
 }
 
+static void _backend_flap_check(struct mcp_backendconn_s *be, enum proxy_be_failures err) {
+    struct timeval now;
+    struct timeval *flap = &be->tunables.flap;
+
+    switch (err) {
+        case P_BE_FAIL_TIMEOUT:
+        case P_BE_FAIL_DISCONNECTED:
+        case P_BE_FAIL_WRITING:
+        case P_BE_FAIL_READING:
+            if (flap->tv_sec != 0 || flap->tv_usec != 0) {
+                struct timeval delta = {0};
+                int64_t subsec = 0;
+                gettimeofday(&now, NULL);
+                delta.tv_sec = now.tv_sec - be->last_failed.tv_sec;
+                subsec = now.tv_usec - be->last_failed.tv_usec;
+                if (subsec < 0) {
+                    delta.tv_sec--;
+                    subsec += 1000000;
+                    delta.tv_usec = subsec;
```

Comment:
should probably leave a comment that this code is weird because "tv_usec must have room for at least [-1, 1000000]", meaning technically we can't guarantee going to -2...